src/UserRepository/Htpasswd.php

```diff
@@ -45,11 +54,15 @@ public function __construct(string $filename, callable $userFactory)
         $this->filename = $filename;
 
         // Provide type safety for the composed user factory.
-        $this->userFactory = function (
+        $this->userFactory = static function (
             string $identity,
             array $roles = [],
             array $details = []
         ) use ($userFactory) : UserInterface {
+            Assert::isList($roles);
```

Comment:
Similar to above: we don't enforce `list<string>` on the roles, so this assertion is too restrictive